script/install.sh

```diff
@@ -1292,11 +1297,28 @@ case $(uname -m) in
         fatal "Unsupported Arch $(uname -m)"
         ;;
 esac
+if [[ ${ARCH_OVERRIDE:+1} ]]; then
+    ARCH=$ARCH_OVERRIDE
+fi
 if [[ $OS-$ARCH = windows-x86_64 && $COMPILER = ldc* ]]; then
     ARCH=x64
 fi
-if [[ $OS-$ARCH = osx-aarch64 && $COMPILER = ldc* ]]; then
-    ARCH=arm64
+if [[ $OS-$ARCH = osx-aarch64 ]]; then
+    if [[ $COMPILER == dmd* ]]; then
+        log "
+DMD does not have builds for macOS on aarch64/arm64 architecture.
+Switching to x86_64 architecture (requires Rosetta).
+"
+        ARCH=x86_64
```

Comment:
Placing this new logic here will miss the logic on [line 1304](https://github.com/dlang/installer/pull/491/files#diff-08b624c4cf3301c03903fe40d86af08880f095510e2858cf6b0fb3cf2a71c770R1304) which makes sure we use `x64` for LDC downloads.

Reply:
Note that line 1312 is only for DMD. And it is only for OS=osx.  Using `x64` for ldc is only relevant for windows.

Reply:
My bad!